### DISCUSSION TITLE
Which reference frame and epoch does adjust.exe report? A reporting discrepancy noted.

### DISCUSSION THREAD

**Discussion by jhaasdyk-au** [GeoscienceAustralia/DynAdjust#200] [Q&A]
The purpose of this github discussion is to:
- record findings, feedback and questions to inform planned discussions with icsm-au/DynAdjust subject matter experts
   i.e. to inform additional user(self)-education, or potential enhancements or defects.
 
All findings and recommendations are my own and may not correctly represent current or planned DynAdjust behaviour.

---

**From where does adjust.exe gather the reference frame (and epoch) used for reporting?**
The ADJ file and the adjust.exe output can differ in the reported epoch.

---

Consider the following examples [skye_tutorial_test_datum.txt](https://github.com/icsm-au/DynAdjust/files/9532242/skye_tutorial_test_datum.txt) (rename as .bat batch file to run). Test# below matches this batch file.
(Note:  The geoid has been chosen to match the source data reference frame.  Choice of geoid does not affect this issue)

<--------------<Import.exe&geoid.exe>, <ADJfile output ---->, <adjust.exe cmd output>
Test#, Source, RefFrame, Geoid------>, RefFrame & Epoch---->, RefFrame & Epoch------>,   
Test1, GDA94-, GDA94---, AUSGeoid09--, GDA94----, 01.01.1994, GDA94----, 01.01.1994,
Test4, GDA94-, GDA2020-, AUSGeoid09--, GDA2020--, 01.01.1994, GDA2020--, 01.01.2020
Note in particular that the **ADJ file and the adjust.exe output** differ in the reported epoch for Test 4.
- The adj file reports the epoch from the input dataset (?? the header from that file?? the measurements ??) 
- The adjust.exe output reports the default epoch of the default reference frame (??)

Further, if reftran is used to set the reference frame and a new epoch, (per Setion 4.7.2) then this reporting discrepancy disappears.
<--------------<Import.exe&geoid.exe>, <reftran.exe options>, <ADJfile output ---->, <adjust.exe cmd output>
Test#, Source, RefFrame, Geoid------>, RefFrame & Epoch---->, RefFrame & Epoch---->, RefFrame & Epoch------>,   
Test7, GDA94-, GDA94---, AUSGeoid09--, GDA94----, 12.06.2035, GDA94----, 12.06.2035, GDA94----, 12.06.2035
Test8, GDA94-, GDA2020-, AUSGeoid09--, GDA94----, 12.06.2035, GDA94----, 12.06.2035, GDA94----, 12.06.2035
Test9, GDA94-, GDA2020-, AUSGeoid09--, GDA2020--, 12.06.2035, GDA2020--, 12.06.2035, GDA2020--, 12.06.2035

----

**For discussion with icsm-au/DynAdjust SME)**
1)	Discrepancy in epoch output for  (adjfile and adjust.exe report different epochs).
2)	Discuss the validity of using reftran to set the epoch of a static datum (e.g. reftran skye_set_GDA2020_AG09_reftran_GDA2020_epoch -r GDA2020 -e 12.06.2035)
This is really a case of mixing and matching what ‘epoch’ is used for. See related discussion [Observation epoch vs reference frame epoch: Explicitly distinguish? · Discussion #195 · icsm-au/DynAdjust (github.com)](https://github.com/icsm-au/DynAdjust/discussions/195)

[Post truncated 12/09 to remove extraneous discussion on Reference Frame documentation]

**Comment by rogerfraser** [comment]
Thanks @jhaasdyk-au for capturing the results of your experiences.  

The inconsistency you're seeing between the adj file and screen output when running Test4 isn't good and should be fixed.  Expect to see a fix in the next release.

**Comment by jhaasdyk-au** [comment]
Thanks Roger for reaching out by email (nominally from related Issue #215)

On further consideration, I've also suggested to you the possibility of requiring more user intervention in order to permit a datum change or to mix and match datums. Currently datum change is implicit in adoption of the default adjustment datum with 'only' a warning.

This note is just to note that email conversation of ~16/05/2023, which may or may not spawn separate issues.
